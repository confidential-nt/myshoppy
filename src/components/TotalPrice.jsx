import React from "react";
import { AiFillPlusCircle } from "react-icons/ai";
import { FaEquals } from "react-icons/fa";

export default function TotalPrice({ productInCarts }) {
  const products = Object.entries(productInCarts).map(([k, v]) => v);
  const totalPrice = products.reduce((acc, cur) => {
    return cur.count * cur.price + acc;
  }, 0);
  const deliveryFee = 3000;

  return (
    <div className="flex flex-col w-10/12 mr-auto ml-auto pt-10">
      <div className="flex justify-around items-center mb-10">
        <div className="flex flex-col items-center">
          <span>상품 총액</span>
          <strong className="text-shoppypink">₩{totalPrice}</strong>
        </div>
        <span>
          <AiFillPlusCircle />
        </span>
        <div className="flex flex-col items-center">
          <span>배송비</span>
          <strong className="text-shoppypink">₩{deliveryFee}</strong>
        </div>
        <span>
          <FaEquals />
        </span>
        <div className="flex flex-col items-center">
          <span>총가격</span>
          <strong className="text-shoppypink">
            ₩{totalPrice + deliveryFee}
          </strong>
        </div>
      </div>
      <button className="bg-shoppypink text-white text-xl pt-1 pb-1">
        주문하기
      </button>
    </div>
  );
}
