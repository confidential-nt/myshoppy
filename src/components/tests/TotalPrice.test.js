import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import TotalPrice from "../TotalPrice";
import { fakeProductInCarts } from "../../tests/products";

describe("TotalPrice", () => {
  it("스냅샷 테스트", () => {
    const component = renderer.create(
      <TotalPrice productInCarts={fakeProductInCarts} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("상품총액을 정확하게 계산해야한다", () => {
    render(<TotalPrice productInCarts={fakeProductInCarts} />);
    expect(screen.getByText("₩120000")).toBeInTheDocument();
  });

  it("총 가격을 정확하게 계산해야한다.", () => {
    render(<TotalPrice productInCarts={fakeProductInCarts} />);
    expect(screen.getByText("₩123000")).toBeInTheDocument();
  });
});
