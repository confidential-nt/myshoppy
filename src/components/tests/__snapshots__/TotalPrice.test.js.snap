// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TotalPrice 스냅샷 테스트 1`] = `
<div
  className="flex flex-col w-10/12 mr-auto ml-auto pt-10"
>
  <div
    className="flex justify-around items-center mb-10"
  >
    <div
      className="flex flex-col items-center"
    >
      <span>
        상품 총액
      </span>
      <strong
        className="text-shoppypink"
      >
        ₩
        120000
      </strong>
    </div>
    <span>
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm192 472c0 4.4-3.6 8-8 8H544v152c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V544H328c-4.4 0-8-3.6-8-8v-48c0-4.4 3.6-8 8-8h152V328c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v152h152c4.4 0 8 3.6 8 8v48z"
        />
      </svg>
    </span>
    <div
      className="flex flex-col items-center"
    >
      <span>
        배송비
      </span>
      <strong
        className="text-shoppypink"
      >
        ₩
        3000
      </strong>
    </div>
    <span>
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 304H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32zm0-192H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
        />
      </svg>
    </span>
    <div
      className="flex flex-col items-center"
    >
      <span>
        총가격
      </span>
      <strong
        className="text-shoppypink"
      >
        ₩
        123000
      </strong>
    </div>
  </div>
  <button
    className="bg-shoppypink text-white text-xl pt-1 pb-1"
  >
    주문하기
  </button>
</div>
`;
